src/metabase/permissions/models/permissions_group_membership.clj

```diff
@@ -35,6 +36,19 @@
       (throw (ex-info (tru "You cannot add or remove users to/from the ''All Users'' group.")
                       {:status-code 400})))))
 
+(def ^:private user-id->is-tenant-user?
```

Comment:
Reminder for myself later: we can't memoize this quite so cleanly because we'll need to move existing users to tenants. :/